src/commands/fate.js

```diff
@@ -0,0 +1,127 @@
+const { makePipe, textMatches } = require('src/commands/util')
+
+const KEY = 'fate'
+
+const regExps = {
+    gainFate: /fate (\w+) gain\s?(\d+)?/i,
+    getCharacter: /fate (\w+) show/i,
+    refresh: /fate (\w+) refresh/i,
+    setRefresh: /fate (\w+) set refresh(?::)? (\d+)/i,
+}
+
+const create = store => store.save(KEY, '[]')
+    .then(() => '[]')
+
+const loadOrCreate = store => store.load(KEY)
+    .catch(() => create(store))
+    .then(JSON.parse)
+
+const save = store =>
+    data =>
+        store
+            .save(KEY, JSON.stringify(data))
+            .then(() => data)
+
+const findCharacter = (character, data) =>
+    data.find(x => x.character === character)
+    || makeCharacter(character)
+
+const findCharacterOrFail = character =>
+    data => {
+        const characterObject = data.find(x => x.character === character)
+        if (!characterObject) throw new Error('Character not found')
+        return characterObject
+    }
+
+const makeCharacter = character => ({
+    character,
```

Comment:
Wouldn't it make more sense to cal the whole entity `character` instead of having `characterObject` variables and call this property `name` or something?

Reply:
Good point.